src/internal/operators/bufferTime.ts

```diff
@@ -69,189 +79,128 @@ export function bufferTime<T>(bufferTimeSpan: number, bufferCreationInterval: nu
  * @return {Observable<T[]>} An observable of arrays of buffered values.
  * @name bufferTime
  */
-export function bufferTime<T>(bufferTimeSpan: number): OperatorFunction<T, T[]> {
-  let length: number = arguments.length;
-
+export function bufferTime<T>(bufferTimeSpan: number, ...otherArgs: any[]): OperatorFunction<T, T[]> {
   let scheduler: SchedulerLike = async;
-  if (isScheduler(arguments[arguments.length - 1])) {
-    scheduler = arguments[arguments.length - 1];
-    length--;
-  }
 
-  let bufferCreationInterval: number | null = null;
-  if (length >= 2) {
-    bufferCreationInterval = arguments[1];
+  if (isScheduler(otherArgs[otherArgs.length - 1])) {
+    scheduler = otherArgs.pop() as SchedulerLike;
   }
 
-  let maxBufferSize: number = Infinity;
-  if (length >= 3) {
-    maxBufferSize = arguments[2];
-  }
+  const bufferCreationInterval = (otherArgs[0] as number) || null;
```

Comment:
Can we use `??` in this case? If `otherArgs[0]` is 0 then this becomes null. Not a big deal in this case but can bite us when we miss it

Reply:
Changed that, and also added a test around scheduling creation intervals at zero -- like some sort of psychopath.